docs/components/index.md

```diff
@@ -0,0 +1,337 @@
+---
+title: Blazor components
+author: guardrex
+description: Learn how to create and use Blazor components, the fundamental building blocks of Blazor apps provided by compiled Razor or C# files.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/16/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/components/index
+---
+# Blazor components
+
+By [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnet/Blazor.Docs/docs/components/common/samples/) ([how to download](xref:client-side/blazor/index#view-and-download-samples)). See the [Get started](xref:client-side/blazor/get-started) topic for prerequisites.
+
+Blazor apps are built using *components*. A component is a self-contained chunk of user interface (UI), such as a page, dialog, or form. A component includes both the HTML markup to render along with the processing logic needed to inject data or respond to UI events. Typically, components are written using Razor files (*\*.cshtml*). Components are flexible and lightweight, and they can be nested, reused, and shared between projects.
+
+## Component classes
+
+Components are implemented using C# classes. Classes are usually saved in *\*.cshtml* files using a combination of HTML and C#.
+
+Dynamic rendering logic (for example, loops, conditionals, expressions) are added using an embedded C# syntax called [Razor](https://docs.microsoft.com/aspnet/core/mvc/views/razor).
+
+When a Blazor app is compiled, the HTML markup and C# rendering logic are converted into a component class. The name of the generated .NET class matches the name of the file.
+
+Members of the component class are defined in a `@functions` block (more than one `@functions` block is permissible). In the `@functions` block, component state (properties, fields) is specified along with methods for event handling or for defining other component logic.
+
+Component members can then be used as part of the component's rendering logic using C# expressions that start with `@`. For example, a C# field is rendered by prefixing `@` to the field name. The following example evaluates and renders:
+
+* `_headingFontStyle` to the CSS property value for `font-style`.
+* `_headingText` to the content of the `<h1>` element.
+
+```cshtml
+<h1 style="font-style:@_headingFontStyle">@_headingText</h1>
+
+@functions {
+    private string _headingFontStyle = "italic";
+    private string _headingText = "Put on your new Blazor!";
+}
+```
+
+## Directives
```

Comment:
How about we rename the "Additional resources" section at the end of this doc to "Razor support" and move this table into that section along with the link to the Razor syntax reference doc.

Reply:
This is here to explain Blazor top-to-bottom matching the layout of a Razor file. The first thing the dev sees looking that these files are the directives. Therefore, I felt that explaining them first (briefly) matched the flow of what a dev sees.

We could have a short directives paragraph here (perhaps just two sentences + a bookmark link) and move the table to the bottom. Do you want to try that approach?

Reply:
The thing is, you don't need any directives initial besides `@functions` to start writing a component. The directives come later for specific scenarios that are covered in their respective sections (DI, usings, routing, etc.)

Reply:
> you don't need any directives ... to start writing a component

... but this is a reference topic . What we usually try to do 😅 in these types of topics is write from general-to-specific and top-to-bottom. This arrangement flips that. We now have general coverage (directives are the most general Razor configuration) at the bottom and top-of-the-file coverage (the first thing a dev sees) at the end of the topic.

I pulled it at the top. I'm happy we're not cutting it entirely. I'm sure devs will see it when they get to the bottom.